File: client/src/schema/timeByTask.ts
import type { MergeCoreCollection } from '../index.js';

/**
 * vytex_compositions type
 */
export type VytexTimeByTask<Schema = any> = MergeCoreCollection<
	Schema,
	'vytex_time-by-task',
	{
		id: number;
		trazar: number;
		plantear: number;
		tender: number;
		cortar: number;
		paquetear: number;
		filetear: number;
		armar: number;
		tapar: number;
		figurar: number;
		marquilla: number;
		cerrar: number;
		gafetes: number;
		presillar: number;
		pulir: number;
		revisar: number;
		acabados: number;
		bolsas: number;
		tiquetear: number;
		empacar: number;
		organizar: number;
		grabar: number;
		paletizar: number;
	}
>;

File: api/app/v1/formats/time_by_task.go
package formats

import "github.com/knovalab-systems/vytex/app/v1/models"

func TimeByTaskDTOFormat(t models.TimeByTaskDTO) *models.TimeByTask {

	return &models.TimeByTask{
		Trazar:    &t.Trazar,
		Plantear:  &t.Plantear,
		Tender:    &t.Tender,
		Cortar:    &t.Cortar,
		Paquetear: &t.Paquetear,
		Filetear:  &t.Filetear,
		Armar:     &t.Armar,
		Tapar:     &t.Tapar,
		Figurar:   &t.Figurar,
		Marquilla: &t.Marquilla,
		Cerrar:    &t.Cerrar,
		Gafetes:   &t.Gafetes,
		Presillar: &t.Presillar,
		Pulir:     &t.Pulir,
		Revisar:   &t.Revisar,
		Acabados:  &t.Acabados,
		Bolsas:    &t.Bolsas,
		Tiquetear: &t.Tiquetear,
		Empacar:   &t.Empacar,
		Organizar: &t.Organizar,
		Grabar:    &t.Grabar,
		Paletizar: &t.Paletizar,
	}
}
